tools/memap.py

```diff
@@ -486,13 +495,13 @@ def main():
 
     parser.add_argument('file', help='memory map file')
 
-    parser.add_argument('-t', '--toolchain', dest='toolchain', help='select a toolchain used to build the memory map file (ARM, GCC_ARM, IAR)',\
-                        required=True)
+    parser.add_argument('-t', '--toolchain', dest='toolchain', help='select a toolchain used to build the memory map file (%s)' % ", ".join(MemapParser.toolchains),\
```

Comment:
This line changes the help output from:

```
-t TOOLCHAIN, --toolchain TOOLCHAIN
                      select a toolchain used to build the memory map file
                      (ARM, GCC_ARM, IAR)
```

to

```
-t TOOLCHAIN, --toolchain TOOLCHAIN
                      select a toolchain used to build the memory map file
                      (ARM, ARM_STD, ARM_MICRO, GCC_ARM, IAR)
```

While agree that these choices should come from our codebase, the listed toolchains should use the public names for the toolchains:

```
-t TOOLCHAIN, --toolchain TOOLCHAIN
                      select a toolchain used to build the memory map file
                      (ARM, uARM, GCC_ARM, IAR)
```